ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -366,13 +367,22 @@ private void acquireLocks() throws CommandProcessorException {
         driverContext.getTxnManager().getTableWriteId(t.getDbName(), t.getTableName());
       }
 
-
       DDLDescWithWriteId acidDdlDesc = driverContext.getPlan().getAcidDdlDesc();
       boolean hasAcidDdl = acidDdlDesc != null && acidDdlDesc.mayNeedWriteId();
```

Comment:
Maybe move this whole stuff to getTxnWriteIds instead?
So we move the whole writeId generation to a single place?

Reply:
Thats more of a refactoring and not sure how it would affect the test cases. I don't want to refactor it as part of this patch, as it's already big and hundreds of test cases are failing.  However, I created a JIRA for the same -> https://issues.apache.org/jira/browse/HIVE-23732 .